relayer/src/foreign_client.rs

```diff
@@ -783,6 +783,36 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
     }
 
     pub fn refresh(&mut self) -> Result<Option<Vec<IbcEvent>>, ForeignClientError> {
+        match self.try_refresh() {
+            Ok(res) => {
+                // If elapsed < refresh_window for the client, `try_refresh()` will
+                // be successful with an empty vector.
+                if let Some(ibc_events) = res.clone() {
```

Comment:
I see, so I removed the `clone()` in commit 47a49b95c32b5cbf481d67d897b8ddba9fc51e6a, but I am not sure I understand how `Option::map` could be used here. Would the idea be to apply the `.map()` to `res` in order to avoid the `Ok(None)` when `res` is `None`?

Reply:
Yes, see if `res.map(|ibc_events| { ... })` could be more elegant here.